generate.py

```diff
@@ -113,11 +114,13 @@ def main(
     with EmptyInitOnDevice(
         device=fabric.device, dtype=dtype, quantization_mode=quantize
     ):
-        model = LLaMA.from_name(model_size)
-
-    checkpoint = lazy_load(checkpoint_path)
-    model.load_state_dict(checkpoint)
-    print(f"Time to load model: {time.time() - t0:.02f} seconds.", file=sys.stderr)
+        print("Loading model ...", file=sys.stderr)
+        t0 = time.time()
+        checkpoint = lazy_load(checkpoint_path)
+        name = llama_model_lookup(checkpoint)
+        model = LLaMA.from_name(name)
+        model.load_state_dict(checkpoint)
+        print(f"Time to load model: {time.time() - t0:.02f} seconds.", file=sys.stderr)
```

Comment:
moving this inside here is reverting https://github.com/Lightning-AI/lit-llama/commit/88ba00804a63684a819fb4960fef778d3900a8ee

so we need to adjust this

Reply:
same applies to the other generate scripts as well.

Reply:
this block of code (or very similar versions) is repeated too many times:

```
    print("Loading model ...", file=sys.stderr)
    t0 = time.time()
    with EmptyInitOnDevice(
        device=fabric.device, dtype=dtype, quantization_mode=quantize
    ):
        model = LLaMA(LLaMAConfig())  # TODO: Support different model sizes

    # 1. Load the pretrained weights
    pretrained_checkpoint = lazy_load(pretrained_path)
    model.load_state_dict(pretrained_checkpoint, strict=False)
```
Can we add it to the utils module, and pass a path, and return the model?

Reply:
OK, I have refactored model loading into utils.load_model
It can handle a regular model by default, or an adapter model when adapter=True.

Reply:
I think we should definitely keep the loading statements in the individual scripts. They are intentionally duplicated for flexibility between the files and there are minor differences that matter. For example the LLaMA model for adapter is not the same as the regular LLaMA. To your question what is "lora", is a context manager that converts the layers in the LLaMA model to LoRA layers corresponding to the finetuned model from finetune_lora.py.

In any case, I think you achieved what was desired for #158 and I would keep further refactoring efforts out of this PR.

Reply:
OK, I have reverted the load function and tidied it up.